src/test/java/uk/gov/pay/directdebit/payments/services/TransactionServiceTest.java

```diff
@@ -132,51 +131,48 @@ public void findChargeForExternalIdAndGatewayAccountId_shouldThrow_ifNoTransacti
+                .withState(AWAITING_DIRECT_DEBIT_DETAILS)
+                .toEntity();
+
+        service.payerCreatedFor(transaction);
+
+        verify(mockedPaymentRequestEventService).registerPayerCreatedEventFor(transaction);
+        verifyZeroInteractions(mockedTransactionDao);
+        assertThat(transaction.getState(), is(AWAITING_DIRECT_DEBIT_DETAILS));
     }
 
     @Test
-    public void findByPaymentRequestExternalIdAndAccountId_shouldUpdateTransactionStateAndRegisterEventWhenReceivingDDDetails() throws Exception {
-        TransactionFixture transactionFixture = TransactionFixture
+    public void payerEditedFor_shouldRegisterAPayerEditedEvent() {
+        Transaction transaction = TransactionFixture
                 .aTransactionFixture()
-                .withState(AWAITING_DIRECT_DEBIT_DETAILS);
-        when(mockedTransactionDao.findTransactionForExternalIdAndGatewayAccountExternalId(transactionFixture.getPaymentRequestExternalId(), gatewayAccountFixture.getExternalId()))
-                .thenReturn(Optional.of(transactionFixture.toEntity()));
-        Transaction newTransaction = service.receiveDirectDebitDetailsFor(gatewayAccountFixture.getExternalId(), transactionFixture.getPaymentRequestExternalId());
-        PaymentRequest paymentRequest = newTransaction.getPaymentRequest();
-        assertThat(newTransaction.getId(), is(notNullValue()));
-        assertThat(paymentRequest.getId(), is(transactionFixture.getPaymentRequestId()));
-        assertThat(paymentRequest.getExternalId(), is(transactionFixture.getPaymentRequestExternalId()));
-        assertThat(paymentRequest.getReturnUrl(), is(transactionFixture.getPaymentRequestReturnUrl()));
-        assertThat(newTransaction.getGatewayAccountId(), is(transactionFixture.getGatewayAccountId()));
-        assertThat(newTransaction.getGatewayAccountExternalId(), is(transactionFixture.getGatewayAccountExternalId()));
-        assertThat(paymentRequest.getDescription(), is(transactionFixture.getPaymentRequestDescription()));
-        assertThat(paymentRequest.getReference(), is(transactionFixture.getPaymentRequestReference()));
-        assertThat(newTransaction.getAmount(), is(transactionFixture.getAmount()));
-        assertThat(newTransaction.getType(), is(transactionFixture.getType()));
-        assertThat(newTransaction.getState(), is(PROCESSING_DIRECT_DEBIT_DETAILS));
-        verify(mockedPaymentRequestEventService).registerDirectDebitReceivedEventFor(newTransaction);
+                .withState(AWAITING_DIRECT_DEBIT_DETAILS)
+                .toEntity();
+
+        service.payerEditedFor(transaction);
+
+        verify(mockedPaymentRequestEventService).registerPayerEditedEventFor(transaction);
+        verifyZeroInteractions(mockedTransactionDao);
+        assertThat(transaction.getState(), is(AWAITING_DIRECT_DEBIT_DETAILS));
+    }
+
+    @Test
+    public void findByPaymentRequestExternalIdAndAccountId_shouldRegisterEventWhenReceivingDDDetails() {
```

Comment:
`DDDetails` is a bit hard to read! The conventional way to handle this (make acronyms into title case like `Http` or `Url`) results in `Dd`, which is rubbish, so maybe the full `DirectDebitDetails` or just `Details`?

Reply:
👍